Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -1068,14 +1068,12 @@ extension BuildPlan {
         }
     }
 
-    /// Traverses the modules graph, computes destination of every module reference and
-    /// provides the data to the caller by means of `onModule` callback. The products
-    /// are completely transparent to this method and are represented by their module dependencies.
+    /// Calls `onModule` for every module in the the package with the modules that depend on it. The products are
+    /// completely transparent to this method and are represented by their module dependencies.
     package func traverseModules(
```

Comment:
If you want to filter out all of duplicate spots where a particular module is used then all you need to do is modify `depthFirstSearch` to not walk into duplicate nodes, this method can stay the same.

Reply:
Updated it to check whether we visited a node already in `successors`.